tqdm/tests/tests_notebook.ipynb

```diff
@@ -0,0 +1,402 @@
+   "source": [
+    "with capture_output() as out:\n",
+    "    for _ in tqdm_notebook(range(5)):\n",
+    "        sleep(0.1)\n",
+    "    assert len(out.outputs)>=1\n",
+    "    widget = out.outputs[0].data\n",
+    "    assert 'HBox(children=(IntProgress(value=0, max=5)' in widget['text/plain']\n",
+    "    widget_id = widget['application/vnd.jupyter.widget-view+json']['model_id']\n",
+    "    all_widgets = Widget.widgets\n",
+    "    assert widget_id in all_widgets\n",
+    "    meta_widget = str(all_widgets[widget_id])\n",
+    "    for info in ['value=5', 'bar_style','success', \"100%\", '5/5']:\n",
+    "        assert info in meta_widget"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## normal with description"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with capture_output() as out:\n",
+    "    for _ in tqdm_notebook(range(5), desc='Test Desc'):\n",
+    "        sleep(0.1)\n",
+    "    assert len(out.outputs)>=1\n",
+    "    widget = out.outputs[0].data\n",
+    "    # PY3 OR PY2\n",
+    "    assert \"description='Test Desc'\" in widget['text/plain'] or \"description=u'Test Desc'\" in widget['text/plain']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## normal with ncols"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO: reopen this after bugfix in python2\n",
+    "\n",
+    "# with capture_output() as out:\n",
+    "#     for _ in tqdm_notebook(range(5), ncols=u'800px'):\n",
+    "#         sleep(0.1)\n",
+    "    \n",
+    "#     widget = out.outputs[0].data\n",
+    "#     widget_id = widget['application/vnd.jupyter.widget-view+json']['model_id']\n",
+    "#     all_widgets = Widget.widgets\n",
+    "#     assert widget_id in all_widgets\n",
+    "#     meta_widget = str(all_widgets[widget_id])\n",
+    "#     assert \"width='800px'\" in meta_widget or \"width=u'800px'\" in meta_widget    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO: reopen this after bugfix in python2\n",
+    "\n",
+    "# with capture_output() as out:\n",
+    "#     for _ in tqdm_notebook(range(5), ncols=800):\n",
+    "#         sleep(0.1)\n",
+    "    \n",
+    "#     widget = out.outputs[0].data\n",
+    "#     widget_id = widget['application/vnd.jupyter.widget-view+json']['model_id']\n",
+    "#     all_widgets = Widget.widgets\n",
+    "#     assert widget_id in all_widgets\n",
+    "#     meta_widget = str(all_widgets[widget_id])\n",
+    "#     assert \"width='800px'\" in meta_widget or \"width=u'800px'\" in meta_widget    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO: reopen this after bugfix in python2\n",
```

Comment:
what bugfix? Can these TODOs be addressed now?

Reply:
Yes, the todos are addressed now. I uncommented them